wagtail/admin/templatetags/wagtailadmin_tags.py

```diff
@@ -648,14 +649,22 @@ def _abs(val):
 def admin_urlquote(value):
     return quote(value)
 
-
 @register.simple_tag
 def avatar_url(user, size=50, gravatar_only=False):
     """
-    A template tag that receives a user and size and return
-    the appropriate avatar url for that user.
+    A template tag that receives a user and size and returns
+    the appropriate avatar URL for that user.
     Example usage: {% avatar_url request.user 50 %}
     """
+    # Custom hook logic
+    for hook_fn in hooks.get_hooks("get_avatar_url"):
+        url = hook_fn(user, size)
+        if url:
+            return url
+
+    # Default logic
+    # Existing logic for gravatar or static avatar URL
+    ...
```

Comment:
I think we can simplify this, the code is clear enough without the comments

```suggestion
    for hook_fn in hooks.get_hooks("get_avatar_url"):
        if url := hook_fn(user, size):
            return url
```

Reply:
Okay Sure Should i commit the new changes that you said?

Reply:
Yes, but I haven't tested it so you'll need to make sure it still works